File: app/src/main/java/com/marlon/portalusuario/database/une/UneDAO.java
package com.marlon.portalusuario.database.une;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.marlon.portalusuario.une.Une;

import java.util.List;

@Dao
public interface UneDAO {

    @Insert
    void insertUne(Une Une);

    @Update
    void updateUne(Une Une);

    @Delete
    void deleteUne(Une Une);

    @Query("SELECT * FROM une")
    LiveData<List<Une>> getAllUne();

    @Query("DELETE FROM une")
    void deleteAllUne();
}

File: app/src/main/java/com/marlon/portalusuario/database/une/UneDAO.kt
package com.marlon.portalusuario.database.une

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.marlon.portalusuario.une.Une
import kotlinx.coroutines.flow.Flow

@Dao
interface UneDAO {
    @Insert
    fun insertUne(une: Une)

    @Update
    fun updateUne(une: Une)

    @Delete
    fun deleteUne(une: Une)

    @Query("SELECT * FROM une")
    fun getAllUne(): Flow<List<Une>>

    @Query("DELETE FROM une")
    fun deleteAllUne()
}
